### PR TITLE
[22317] Test service node listener

### PR DESCRIPTION
**Pull request by Javgilavi**
This PR add the blackbox tests for service node listener and the minimum needed implementation.

### CODE REVIEW

File: sustainml_cpp/test/blackbox/api/ManagedNodeService.hpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+#include <sustainml_cpp/nodes/AppRequirementsNode.hpp>
+#include <sustainml_cpp/nodes/CarbonFootprintNode.hpp>
+#include <sustainml_cpp/nodes/HardwareConstraintsNode.hpp>
+#include <sustainml_cpp/nodes/HardwareResourcesNode.hpp>
+#include <sustainml_cpp/nodes/MLModelNode.hpp>
+#include <sustainml_cpp/nodes/MLModelMetadataNode.hpp>
+#include <sustainml_cpp/core/RequestReplyListener.hpp>
+
+template <typename _NODE_TYPE, typename _LISTENER_TYPE, typename ... Args>
+class ManagedNodeService
```
Comment (Mario-DL) [inline sustainml_cpp/test/blackbox/api/ManagedNodeService.hpp:34]: Instead of maintaining this new `ManagedNodeService` class and the old `ManagedNode` , extend the old `ManagedNode` including the logic of this new one (which is basically adding the `ManagedRequestReplyListener`)

**Review by Mario-DL — CHANGES_REQUESTED** [review]
This test suite will gracefully assert the feature's correct behavior, good work.
Leaving some suggestions though

**Comment by Mario-DL** [comment]
Additionally, we should uncrustify the `types.cpp` (which is not generated code) and `OrchestratorNode.hpp` as the ci reports code divergences on them

File: sustainml_cpp/include/sustainml_cpp/core/RequestReplyListener.hpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (Mario-DL) [inline sustainml_cpp/include/sustainml_cpp/core/RequestReplyListener.hpp:1]:
```suggestion
// Copyright 2025 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

**Review by Mario-DL — CHANGES_REQUESTED** [review]

File: sustainml_cpp/include/sustainml_cpp/orchestrator/OrchestratorNode.hpp [hunk 3]
```diff
@@ -174,6 +174,14 @@ class OrchestratorNode
     void send_control_command(
             const types::NodeControl& cmd);
 
+     /**
+     * @brief This method sends a request to node via service for changing its configuration.
+     * @param [in] req configuration request, contain which node and configuration file
+     * @param [out] res response from the node with its new configuration
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply sustainml_cpp/include/sustainml_cpp/orchestrator/OrchestratorNode.hpp]: I think this suggestion was not applied, I am missing the `return` in the doxygen comment

**Review by Mario-DL — APPROVED** [review]
LGTM !
